### ISSUE TITLE
Stresstest is failing upon start up

### ISSUE THREAD

**Issue by fredo** [raiden-network/raiden#5948]
I think these are recent changes due to the introduction of the nursery and janitor. 

Obviuously, this part of the code creates the error message if the try fail: 

```
                    try:
                        process = Popen(args)
                    finally:
                        janitor._processes.add(process)
```

### Traceback

```
Traceback (most recent call last):
  File "/home/fred/fred/projects/raiden/raiden/raiden/utils/nursery.py", line 95, in exec_under_watch
    process = Popen(args)
  File "/home/fred/fred/projects/raiden/raiden/venv/lib/python3.7/site-packages/gevent/subprocess.py", line 656, in __init__
    restore_signals, start_new_session)
  File "/home/fred/fred/projects/raiden/raiden/venv/lib/python3.7/site-packages/gevent/subprocess.py", line 1378, in _execute_child
    args = list(args)
TypeError: 'Popen' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 854, in gevent._greenlet.Greenlet.run
  File "/home/fred/fred/projects/raiden/raiden/tools/debugging/stress_test_transfers.py", line 213, in start_and_wait_for_server
    nursery.exec_under_watch(process)
  File "/home/fred/fred/projects/raiden/raiden/raiden/utils/nursery.py", line 97, in exec_under_watch
    janitor._processes.add(process)
UnboundLocalError: local variable 'process' referenced before assignment
2020-02-27T14:54:08Z <Greenlet at 0x7f834c16ad08: start_and_wait_for_server(<raiden.utils.nursery.Janitor.__enter__.<locals>.P, <generator object _unused_ports at 0x7f834bef6d68>, NodeConfig(args=['raiden', '--accept-disclaimer', , 1)> failed with UnboundLocalError

Traceback (most recent call last):
  File "/home/fred/fred/projects/raiden/raiden/raiden/utils/nursery.py", line 95, in exec_under_watch
    process = Popen(args)
  File "/home/fred/fred/projects/raiden/raiden/venv/lib/python3.7/site-packages/gevent/subprocess.py", line 656, in __init__
    restore_signals, start_new_session)
  File "/home/fred/fred/projects/raiden/raiden/venv/lib/python3.7/site-packages/gevent/subprocess.py", line 1378, in _execute_child
    args = list(args)
TypeError: 'Popen' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "src/gevent/greenlet.py", line 854, in gevent._greenlet.Greenlet.run
  File "/home/fred/fred/projects/raiden/raiden/tools/debugging/stress_test_transfers.py", line 213, in start_and_wait_for_server
    nursery.exec_under_watch(process)
  File "/home/fred/fred/projects/raiden/raiden/raiden/utils/nursery.py", line 97, in exec_under_watch
    janitor._processes.add(process)
UnboundLocalError: local variable 'process' referenced before assignment
2020-02-27T14:54:08Z <Greenlet at 0x7f834c16abf8: start_and_wait_for_server(<raiden.utils.nursery.Janitor.__enter__.<locals>.P, <generator object _unused_ports at 0x7f834bef6d68>, NodeConfig(args=['raiden', '--accept-disclaimer', , 1)> failed with UnboundLocalError

Traceback (most recent call last):
  File "/home/fred/fred/projects/raiden/raiden/raiden/utils/nursery.py", line 95, in exec_under_watch
    process = Popen(args)
  File "/home/fred/fred/projects/raiden/raiden/venv/lib/python3.7/site-packages/gevent/subprocess.py", line 656, in __init__
    restore_signals, start_new_session)
  File "/home/fred/fred/projects/raiden/raiden/venv/lib/python3.7/site-packages/gevent/subprocess.py", line 1378, in _execute_child
    args = list(args)
TypeError: 'Popen' object is not iterable

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/fred/fred/projects/raiden/raiden/tools/debugging/stress_test_transfers.py", line 656, in <module>
    main()
  File "/home/fred/fred/projects/raiden/raiden/tools/debugging/stress_test_transfers.py", line 630, in main
    nursery, port_generator, nodes_config, retry_timeout
  File "/home/fred/fred/projects/raiden/raiden/tools/debugging/stress_test_transfers.py", line 231, in start_and_wait_for_all_servers
    gevent.joinall(greenlets, raise_error=True)
  File "src/gevent/greenlet.py", line 1002, in gevent._greenlet.joinall
  File "src/gevent/greenlet.py", line 1018, in gevent._greenlet.joinall
  File "src/gevent/greenlet.py", line 364, in gevent._greenlet.Greenlet._raise_exception
  File "/home/fred/fred/projects/raiden/raiden/venv/lib/python3.7/site-packages/gevent/_compat.py", line 62, in reraise
    raise value.with_traceback(tb)
  File "src/gevent/greenlet.py", line 854, in gevent._greenlet.Greenlet.run
  File "/home/fred/fred/projects/raiden/raiden/tools/debugging/stress_test_transfers.py", line 213, in start_and_wait_for_server
    nursery.exec_under_watch(process)
  File "/home/fred/fred/projects/raiden/raiden/raiden/utils/nursery.py", line 97, in exec_under_watch
    janitor._processes.add(process)
UnboundLocalError: local variable 'process' referenced before assignment
```

**Comment by karlb** [comment]
See also https://github.com/raiden-network/raiden/pull/5960

**Comment by hackaugusto** [comment]
this was fixed by 706b7e67293dd40f870fb659b9fa87dcc59a8340
